src/main/java/ch/naviqore/service/gtfs/raptor/convert/GtfsToRaptorConverter.java

```diff
@@ -99,24 +99,46 @@ private void addRoute(GtfsRoutePartitioner.SubRoute subRoute) {
      */
     private void processAllTransfers() {
         addAdditionalTransfers();
-        processStopAndParentChildTransfers();
-        addGtfsTransfersWithPrecedence();
+        // It is possible that stops are added without departures in this loop, in that case this block will be
+        // re-run to ensure that transfers from those newly added stops are also included.
+        while (!stopsForTransfers.isEmpty()) {
+            Set<String> stopIterator = new HashSet<>(stopsForTransfers);
+            stopsForTransfers.clear();
+            processStopAndParentChildTransfers(stopIterator);
+            addGtfsTransfersWithPrecedence(stopIterator);
+        }
     }
 
     /**
      * Adds all additional transfers.
      */
     private void addAdditionalTransfers() {
         for (TransferGenerator.Transfer transfer : additionalTransfers) {
-            builder.addTransfer(transfer.from().getId(), transfer.to().getId(), transfer.duration());
+            this.addTransfer(transfer.from().getId(), transfer.to().getId(), transfer.duration());
         }
     }
 
+    private void addStop(String stopId) {
```

Comment:
Do we still have the inconsistency between the GTFS schedule and the RAPTOR Router? Specifically, there could be stops in the schedule that the router doesn't recognize. I assume in that case, the router throws an exception when a route is requested from a stop with no departures in the GTFS.

Reply:
If we decide to add all stops to the router, it would simply result in no connections being returned, which accurately reflects the reality: the stop exists in the schedule, but there are no departures and, therefore, no connections.

Whether it makes semantic sense to include stops without departures in a schedule is another question entirely, but I believe that is not our responsibility to address.